tests/geo/spec.cc

```diff
@@ -0,0 +1,381 @@
+            {C({{"area", v{90, -180, -90, 180}}}), BAD},
+            {C({{"grid", "B48"}}), BAD},
+            {C({{"grid", "F48"}}), C({{"type", "regular_gg"}})},
+            {C({{"grid", "N48"}}), C({{"type", "reduced_gg"}})},
+            {C({{"grid", "O48"}}), C({{"type", "reduced_gg"}})},
+            {C({{"grid", 48}}), BAD},
+            {C({{"grid", v{2, 2}}}), C({{"grid", v{2, 2}}, {"type", "regular_ll"}})},
+            {C({{"type", "latlon"}, {"grid", v{2, 2}}}), BAD},
+            {C({{"type", "reduced_gg"}, {"grid", "48"}}), BAD},
+            {C({{"type", "reduced_gg"}, {"grid", "F048"}}), BAD},
+            {C({{"type", "reduced_gg"}, {"grid", "N"}}), BAD},
+            {C({{"type", "reduced_gg"}, {"grid", "N048"}}), BAD},
+            {C({{"type", "reduced_gg"}, {"grid", "N48"}}), C({{"type", "reduced_gg"}})},
+            {C({{"type", "reduced_gg"}, {"grid", "O048"}}), BAD},
+            {C({{"type", "reduced_gg"}, {"grid", "O48"}}), C({{"type", "reduced_gg"}})},
+            {C({{"type", "reduced_gg"}, {"grid", 48}}), BAD},
+            {C({{"type", "reduced_gg"}}), BAD},
+            {C({{"type", "reduced_latlon"}, {"grid", 2}}), BAD},
+            {C({{"type", "reduced_ll"}, {"grid", 12}}), BAD},
+            {C({{"type", "regular_gg"}, {"grid", "48"}}), BAD},
+            {C({{"type", "regular_gg"}, {"grid", "F048"}}), BAD},
+            {C({{"type", "regular_gg"}, {"grid", "F48"}}), C({{"type", "regular_gg"}})},
+            {C({{"type", "regular_gg"}, {"grid", "N48"}}), BAD},
+            {C({{"type", "regular_gg"}, {"grid", "O48"}}), BAD},
+            {C({{"type", "regular_gg"}, {"grid", "a"}}), BAD},
+            {C({{"type", "regular_gg"}, {"grid", 48}}), BAD},
+            {C({{"type", "regular_ll"}, {"area", v{90, -180, -90, 180}}}), BAD},
+            {C({{"type", "regular_ll"}, {"grid", "F48"}}), BAD},
+            {C({{"type", "regular_ll"}, {"grid", "a"}}), BAD},
+            {C({{"type", "regular_ll"}, {"grid", 48}}), BAD},
+            {C({{"type", "regular_ll"}, {"grid", std::vector<std::string>{"a", "b"}}}), BAD},
+            {C({{"type", "regular_ll"}, {"grid", v{1, 2, 3}}}), BAD},
+            {C({{"type", "regular_ll"}, {"grid", v{1, 2}}}), C({{"type", "regular_ll"}})},
+            {C({{"type", "regular_ll"}, {"grid", v{1}}}), BAD},
+
+            {C({{"type", "mercator"},
+                {"area", v{31.173058, 262.036499, 14.736453, 284.975281}},
+                {"grid", v{45000.0, 45000.0}},
+                {"shape", std::vector<size_t>{56, 44}},
+                {"lad", 14.0},
+                {"orientation", 0.0}}),
+             C()},
+        };
+
+        for (const auto& [user, gridspec] : tests) {
+            Log::info() << user << " -> " << gridspec << std::endl;
+
+            try {
+                std::unique_ptr<const Spec> spec(GridFactory::make_spec(user));
+                EXPECT(spec);
+
```

Comment:
This code should probably test the contents of the generated `spec` not only its existence.

Reply:
Yes, that was a missing test until the generation of grid specs was at all possible, I will add this.